### DISCUSSION TITLE
Support for time varying topology (as in Vpaint)

### DISCUSSION THREAD

**Discussion by vpaintanimation** [opentoonz/opentoonz#6496] [Ideas]
Similar to vpaint.  The open source protocol it uses seems to be compatible with opentoonz.  I hope opentoonz can be like it https://youtu.be/Xk1_CugdytI.  I hope in opentoonz. point and line and face can transform each other at will. Time-varying topology .It may require opentoonz to have a concept of 3D matrix space.

**Comment by vpaintanimation** [comment]
This is an example of an animation I made.  Used to show the topological changes of points and line and face.  Intermediate frames are automatically generated
[11.zip](https://github.com/opentoonz/opentoonz/files/5705786/11.zip)

**Comment by artisteacher** [comment]
That looks like it could be helpful.

**Comment by ghost** [comment]
The inbetween generation feature of vpaint looks very interesting.I wonder if the vpaint genetation feature would be compatible with the current auto inbetweener generation of OT, (e.f the difficulties of the curve animation PR #1878.). It seems that could be to fix somes issues mentioned in #3341 about retain strokes and spacing issues.

**Comment by artisteacher** [comment]
If this could be paired with the inbetweener, or vector guided drawing commands, #1423 probably wouldn't be necessary.

**Comment by vpaintanimation** [comment]
Cacani's auto inbetween frame animation function.  Only the line is deformed.  In two key frames.  The number of strokes needs to correspond one to one. If opentoonz supports spatio-temporal topology changes. This means.  A line can become multiple lines time-varying. One face becomes multiple faces.  The vertex is too.  And the three of them can be arbitrarily transformed between two keyframes. This will greatly improve the efficiency of animation production.  The premise is.  It may require opentoonz to have a 3D matrix model as the spatiotemporal data of the canvas and timeline.  The matrix points are used as uniform sampling points.  It is point, the basis of line and face
![Screenshot_20201217_223914](https://user-images.githubusercontent.com/25269197/102502264-5c80bb80-40b9-11eb-8384-5de14f447d98.jpg)

**Comment by vpaintanimation** [comment]
![Screenshot_20201217_225905](https://user-images.githubusercontent.com/25269197/102504542-efbaf080-40bb-11eb-9ba1-217a279e6491.jpg)
![Screenshot_20201217_230136](https://user-images.githubusercontent.com/25269197/102504566-f77a9500-40bb-11eb-8296-095276c8a0fe.jpg)

**Comment by RodneyBaker** [comment]
If any programmers are willing and able to develop this or other vector based features please let the Opentoonz community know of your interest and we will gladly welcome you aboard.  :)

**Comment by RodneyBaker** [comment]
VGC now has a discord for those interested in adding it to their digital art and animation ecosystem... as well as develop the future vector graphics through Opentoonz and VGC.

Link:  https://discord.gg/yA3AaMQV7u

Added:
As this implementation will take considerable time a shorter term goal that would allow Opentoonz users to take advantage of time varying topology would be to include vPaint (and eventually VGC) in their pipeline/workflow.

I confirm that vPaints (Beta) SVG Import/Export works well with Opentoonz.
TODO:  Opentoonz still needs some foundational fixes to SVG conversion (naming placement and scale) in order to establish a smooth workflow.

In short, Opentoonz users can (currently) best leverage the features of vPaint/VCG by actually using those programs.
